graal-js/src/com.oracle.truffle.js.parser/src/com/oracle/truffle/js/parser/GraalJSTranslator.java

```diff
@@ -693,6 +695,12 @@ private JavaScriptNode toGeneratorBlockNode(AbstractBlockNode blockNode, BitSet
             }
             genBlock = returnsResult ? factory.createGeneratorExprBlock(chunks, readState, writeState) : factory.createGeneratorVoidBlock(chunks, readState, writeState);
         }
+        if (blockNode.hasTag(StandardTags.RootBodyTag.class)) {
+            genBlock.addRootBodyTag();
```

Comment:
Why do you need this tag? `RootBodyTag` is for function roots

Reply:
This check here is not necessary, I have updated the code.